packages/client/src/helpers/provideAuthHeaders.ts

```diff
@@ -0,0 +1,27 @@
+import type { Authentication } from '../authentication';
+
+type Handler<Val> = (headers: Record<string, string>) => Promise<Val>;
+
+async function getHeader(
+  authentication: Authentication | undefined,
+): Promise<Record<string, string>> {
+  if (!authentication) {
+    return {};
+  }
+
+  const headerResult = await authentication.getRequestHeader();
+  if (headerResult.isFailure()) {
+    return {};
+  }
+
+  return headerResult.value;
+}
+
+export async function provideAuthHeaders<Val>(
+  authentication: Authentication | undefined,
+  handler: Handler<Val>,
+): Promise<Val> {
+  const header = await getHeader(authentication);
+
+  return handler(header);
```

Comment:
I could export only `getHeader` function (with a better name) but I wanted to keep the usage the same as in `requireAuthHeader` for consistency, but we can discuss that